utils_cv/tracking/model.py

```diff
@@ -1,189 +1,240 @@
-        trainer = Trainer(self.opt, self.model, optimizer)
+        trainer = Trainer(self.opt, self.model, self.optimizer)
         trainer.set_device(
             self.opt.gpus, self.opt.chunk_sizes, self.opt.device
         )
-        best = 1e10
 
         # training loop
-        for epoch in range(start_epoch + 1, self.opt.num_epochs + 1):
+        for epoch in range(self.start_epoch + 1, self.opt.num_epochs + 1):
             mark = epoch if self.opt.save_all else "last"
             log_dict_train, _ = trainer.train(epoch, train_loader)
             ## TODO logging
-            if (
-                self.opt.val_intervals > 0
-                and epoch % self.opt.val_intervals == 0
-            ):
-                save_model(
-                    osp.join(self.opt.save_dir, f"model_{mark}.pth"),
-                    epoch,
-                    model,
-                    optimizer,
-                )
-            else:
-                save_model(
-                    osp.join(
-                        self.opt.save_dir,
-                        f"model_last.pth",
-                        epoch,
-                        model,
-                        optimizer,
-                    )
-                )
-
-        for epoch in self.opt.lr_step:
-            save_model(
-                os.path.join(self.opt.save_dir, f"model_{epoch}.pth"),
-                epoch,
-                model,
-                optimizer,
-            )
-            lr = self.opt.lr * (0.1 ** self.opt.lr_step.index(epoch) + 1)
-            ## TODO logging
-            for param_group in optimizer.param_groups:
-                param_group["lr"] = lr
-
-        if epoch % 5 == 0:  ## TODO make this a param
-            save_model(
-                osp.join(
-                    self.opt.save_dir,
-                    f"model_{epoch}.pth",
-                    epoch,
-                    model,
-                    optimizer,
-                )
-            )
-
-    def _init_dataset(self, data_root, train_path, im_size) -> Dataset:
-        Dataset = get_dataset(self.opt.dataset, self.opt.task)
-        transforms = T.Compose([T.ToTensor()])
-        return Dataset(
-            self.opt,
-            data_root,
-            train_path,
-            im_size,
-            augment=True,
-            transforms=transforms,
-        )
+            if epoch in self.opt.lr_step:
+                lr = self.opt.lr * (0.1 ** self.opt.lr_step.index(epoch) + 1)
+                ## TODO logging
+                for param_group in optimizer.param_groups:
+                    param_group["lr"] = lr
 
     def predict(
         self,
-        im_or_video_path: Path,
+        im_or_video_path: str,
         conf_thres: float = 0.6,
         det_thres: float = 0.3,
         nms_thres: float = 0.4,
         track_buffer: int = 30,
         min_box_area: float = 200,
-        input_w: float = -1,
         input_h: float = -1,
+        input_w: float = -1,
```

Comment:
[minor] here and other places: could we use None as indicator rather than -1.

Reply:
i kept -1 because that's what the FairMOT source code uses. If using None is more in-line with CVBP, I can change our TrackingLearner interface to take None but FairMOT src will still use -1 under the hood

Reply:
Makes sense. Yeah, would be good to make it None for CVBP, and similar also at other places where -1 is used.